model/jpa/src/main/resources/META-INF/jpa-changelog-authz-3.4.0.CR1.xml

```diff
@@ -28,13 +28,32 @@
         <addColumn tableName="RESOURCE_SERVER_SCOPE">
             <column name="RESOURCE_SERVER_CLIENT_ID" type="VARCHAR(36)"/>
         </addColumn>
+    </changeSet>
 
+    <changeSet author="glavoie@gmail.com" id="authz-3.4.0.CR1-resource-server-pk-change-part2">
+        <preConditions onSqlOutput="TEST" onFail="MARK_RAN">
+            <not>
+                <dbms type="mssql" />
+            </not>
+        </preConditions>
         <sql>
-            UPDATE RESOURCE_SERVER_POLICY p SET RESOURCE_SERVER_CLIENT_ID=(SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
-            UPDATE RESOURCE_SERVER_RESOURCE p SET RESOURCE_SERVER_CLIENT_ID=(SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
-            UPDATE RESOURCE_SERVER_SCOPE p SET RESOURCE_SERVER_CLIENT_ID=(SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
+            UPDATE RESOURCE_SERVER_POLICY p SET RESOURCE_SERVER_CLIENT_ID = (SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
+            UPDATE RESOURCE_SERVER_RESOURCE p SET RESOURCE_SERVER_CLIENT_ID = (SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
+            UPDATE RESOURCE_SERVER_SCOPE p SET RESOURCE_SERVER_CLIENT_ID = (SELECT CLIENT_ID FROM RESOURCE_SERVER s WHERE s.ID = p.RESOURCE_SERVER_ID);
         </sql>
+    </changeSet>
+    <changeSet author="glavoie@gmail.com" id="authz-3.4.0.CR1-resource-server-pk-change-part2-mssql">
+        <preConditions onSqlOutput="TEST" onFail="MARK_RAN">
+            <dbms type="mssql" />
```

Comment:
@hmlnarik, after some testing, my DBA told me the Oracle syntax works on both PostgreSQL and MySQL, but the MSSQL syntax only on PostgreSQL. I did the exception for MSSQL. DB2 remains.

Reply:
There is DB2 result: 
ERROR [org.keycloak.connections.jpa.updater.liquibase.conn.DefaultLiquibaseConnectionProvider] (ServerService Thread Pool -- 50) Change Set META-INF/jpa-changelog-authz-3.4.0.CR1.xml::authz-3.4.0.CR1-resource-server-pk-change-part3::glavoie@gmail.com failed.  Error: DB2 SQL Error: SQLCODE=-478, SQLSTATE=42893, SQLERRMC=COLUMN;PK_RESOURCE_SERVER;PRIMARY KEY, DRIVER=4.19.26 [Failed SQL: ALTER TABLE DBALLO14.RESOURCE_SERVER RENAME COLUMN CLIENT_ID TO ID]: liquibase.exception.DatabaseException: DB2 SQL Error: SQLCODE=-478, SQLSTATE=42893, SQLERRMC=COLUMN;PK_RESOURCE_SERVER;PRIMARY KEY, DRIVER=4.19.26 [Failed SQL: ALTER TABLE DBALLO14.RESOURCE_SERVER RENAME COLUMN CLIENT_ID TO ID]

Reply:
@vramik, pushed a new version that drops all FK/PK/indices, renames the columns, then adds everything back. Still works on MySQL/MSSQL/Oracle. Any better on DB2?